Engine/ac/sys_events.cpp

```diff
@@ -498,9 +498,66 @@ int ags_check_mouse_wheel() {
 // TOUCH INPUT
 // ----------------------------------------------------------------------------
 
+// SDL_FingerIDs are unique for down-motion-up events, but beyond this there is no cross-platform guarantee
+// this struct tracks currently active fingers on the touch screen, and matches a slot index for each
+// we will use this to implement the logic in this example
+// - first finger down (consider LMB, index 0);
+// - second finger down (consider RMB, index 1);
+// - first finger up (consider LMB up);
+// - second finger is still pressed and still considered RMB;
+// - third(?) finger down, takes the free role of LMB (index 0).
+struct Fingers
+{
+public:
+    static const int MAX_FINGERS = 2;
+    static const int NO_ID = -1;
+
+    void push(SDL_FingerID fingerId)
+    {
+        if (contains(fingerId))
+            return;
+
+        for (int i = 0; i < MAX_FINGERS; i++) {
+            if (_fingers[i] == NO_ID) {
+                _fingers[i] = fingerId;
+                return;
+            }
+        }
+    };
+
+    int get_index(SDL_FingerID fingerId) const
+    {
+        for(int i=0; i < MAX_FINGERS; i++) {
+            if(_fingers[i] == fingerId)
```

Comment:
Same as above, this may be replaced with std::find(_fingers.begin(), _fingers.end(), fingerId)

Reply:
how do I return the index using this?

Reply:
> how do I return the index using this?

`it - _fingers.begin()`

Reply:
ok, I tried to do this in the sketch commit below